indico/modules/events/contributions/controllers.py

```diff
@@ -34,4 +56,55 @@ class RHManageContributions(RHManageContributionsBase):
     """Display contributions management page"""
 
     def _process(self):
-        return WPManageContributions.render_template('management/contributions.html', self.event)
+        contribs = Contribution.find_all(event_id=self.event.id, is_deleted=False)
+        sessions = Session.find_all(event_id=self.event.id, is_deleted=False)
+        tracks = self.event.getTrackList()
+        return WPManageContributions.render_template('management/contributions.html', self.event, event=self.event,
+                                                     contribs=contribs, sessions=sessions, tracks=tracks)
+
+
+class RHManageContributionBase(RHManageContributionsBase):
+    """Base class for a specific contribution"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageContributionsBase._checkParams(self, params)
+        self.contrib = Contribution.get(request.view_args['contrib_id'])
+
+
+class RHManageContributionCreate(RHManageContributionsBase):
+    def _process_GET(self):
+        form = ContributionForm()
+        return jsonify_template('events/contributions/management/contrib_dialog.html', form=form)
+
+    def _process_POST(self):
+        form = ContributionForm()
+        if form.validate_on_submit():
```

Comment:
if validation fails you won't return anything. for the normal form-based flow using the two separate methods is really not a good idea - you want to return the template after POST in case validation fails